test/unit/sql/where.test.js

```diff
@@ -821,6 +847,18 @@ suite(Support.getTestDialectTeaser('SQL'), () => {
           sqlite: "(json_extract(`User`.`data`, '$.name.last') = 'Simpson' AND json_extract(`User`.`data`, '$.employment') != 'None')"
         });
 
+        testsql('data', {
+          price: 5,
+          name: 'Product'
```

Comment:
Could you add unit tests for boolean and date matching as well

Reply:
Do you mean to duplicate each $between and $in tests to include all possible casting types ?
Cause we have tests for those castings in Line 934 and 949.

I added the $between and $in tests because previous logic didn't apply casting at all to them

Reply:
> Cause we have tests for those castings in Line 934 and 949.

Roger that!